src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -235,6 +239,31 @@ public void writeLogTo(XMLOutput out) throws IOException {
         @Inject
         private transient GitHubHookRegisterProblemMonitor monitor;
 
+        @Inject
+        private transient SCMTrigger.DescriptorImpl scmTrigger;
+
+        private transient int maximumThreads = Integer.MIN_VALUE;
+
+        public DescriptorImpl() {
+            checkThreadPoolSize();
+        }
+
+        /**
+         * Update the {@link java.util.concurrent.ExecutorService} instance.
+         */
+        /*package*/
+        synchronized void checkThreadPoolSize() {
+            if (scmTrigger != null) {
+                int count = scmTrigger.getPollingThreadCount();
+                if (maximumThreads != count) {
+                    queue.setExecutors(
+                            (count == 0
+                                    ? Executors.newCachedThreadPool(threadFactory())
+                                    : Executors.newFixedThreadPool(maximumThreads, threadFactory())));
```

Comment:
Does this not cause a thread leak?
Say `ScmTrigger.pollingThreadCount` was set to `100`, this kicks in and creates a new fixed size thread pool, then you have a lot of notifcations and you use all 100 Threads...
then the admin sees this and sets `ScmTrigger.pollingThreadCount` to a more sane value of `10`, and this creates a new fixed size thread pool Executor...

But the original ExecutorService (with 100 threads) is not shutdown, and not does it wait for all the tasks to complete...

Basically you are waiting for GC to kick (and hoping that it does the right thing)...